kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -248,14 +251,25 @@ public void shutDown() {
     public static class Builder<K, V> {
 
         private ProducerFactory<K, V> producerFactory;
-        @SuppressWarnings({"unchecked", "squid:S1874"})
-        private KafkaMessageConverter<K, V> messageConverter =
-                (KafkaMessageConverter<K, V>) DefaultKafkaMessageConverter.builder()
-                                                                          .serializer(XStreamSerializer.defaultSerializer())
-                                                                          .build();
+        private KafkaMessageConverter<K, V> messageConverter;
         private MessageMonitor<? super EventMessage<?>> messageMonitor = NoOpMessageMonitor.instance();
         private TopicResolver topicResolver = m -> Optional.of(DEFAULT_TOPIC);
         private long publisherAckTimeout = 1_000;
+        private Supplier<Serializer> serializer = () -> XStreamSerializer.builder()
+                                                                         .xStream(new XStream(new CompactDriver()))
+                                                                         .build();
+
+        /**
+         * Sets the {@link Serializer} used to serialize and deserialize messages.
```

Comment:
It would be friendly if this little bit of Javadoc also shared what the default serializer is.